packages/labs/motion/src/animate.ts

```diff
@@ -534,7 +534,11 @@ export class Animate extends AsyncDirective {
     frames: Keyframe[] | undefined,
     options = this.options.keyframeOptions
   ) {
-    this.start();
+    // Checking existence of frames to prevent situation 
+    // where options.onStart is called and options.onComplete is not. 
+    if (frames !== undefined) {
```

Comment:
I think it probably makes sense to move this into the condition below line 544 with the same check.

The design and order of the callbacks still needs some refinement but that's probably out of scope here. In particular the frames could become undefined via `onFrames` and you'd still get a start without a complete. To fix that, we'd probably want to change the callback ordering so `onFrames` came first and maybe would be called `onPrepare`. That's a breaking change so would need more consideration.

I think here either punting on that case or changing the `didAnimate` flag to `didStart` and setting that if `onStart` runs would both be ok.